### PR TITLE
[Doc] Explain why `thread::yield_now` could be used.

### PR DESCRIPTION
**Pull request by gamazeps**
Part of #29378.

r? @steveklabnik

### CODE REVIEW

File: src/libstd/thread/mod.rs [hunk 1]
```diff
@@ -468,13 +468,35 @@ pub fn current() -> Thread {
 
 /// Cooperatively gives up a timeslice to the OS scheduler.
 ///
+/// This is used when the programmer knows that the thread will have nothing
+/// to do for some time, and thus avoid wasting computing time.
+///
+/// For example when polling on a resource, it is common to check that it is
+/// available, and if not to yield in order to avoid busy waiting. Thus the
+/// poll then yield, is a rather common pattern when implementing low-level
```
Comment (GuillaumeGomez) [inline src/libstd/thread/mod.rs]: This sentence is a bit confusing. "the poll then yield" for example is a bit hard to understand with just one reading. Don't hesitate to make the sentence longer if it helps the reader to understand more easily.

Comment (gamazeps) [inline src/libstd/thread/mod.rs]: Done :)

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors: r+ rollup

**Comment by Mark-Simulacrum** [comment]
Looks like tests just... hung? Stopped? I can't really tell. Either way, there's practically no way for that to be the fault of this PR.

@bors retry

**Comment by Mark-Simulacrum** [comment]
```
test malformed_override ... ok
test unused_keys ... ok
test both_git_and_path_specified has been running for over 60 seconds
```

@bors retry
  - Maybe #40474, but maybe not. Cargo-related.

**Pull request merged**